### DISCUSSION TITLE
Github CI Pipeline issue with \Orchestra\Testbench\Dusk\Options::withoutUI();

### DISCUSSION THREAD

**Discussion by JonathanAspeling** [livewire/livewire#6886] [1. Help] [answered]
Good day! 

I trust you are all doing well! Received a recent issue while running our PEST PHP tests on a Laravel with Livewire build. 

![image](https://github.com/livewire/livewire/assets/42679894/0af198a4-3583-4b92-ae8c-dc93463ed3c3)

Wanted to check if anyone else might have experienced this or if you may have a few ideas on how I can resolve. Suspect it is something on the container or with CI actions themselves. We haven't changed dependencies and this looks to be something random!

I'm still digging around for more info but would be awesome if someone notes this question and has the needed guidance to offer

**Comment by JonathanAspeling** [comment]
Looks like some dependency issues! 

This method referenced here looks to be introduced in V8 of the Orchestra Package. The V8 requires Laravel 10 so if you're on Laravel 9 still and CI spins up then it can't install V8, and the call here fails without the needed V8 version which then blocks the Automated test step of the CI pipeline. 

Composer feedback trying to manually install V8+
![image](https://github.com/livewire/livewire/assets/42679894/ae987f53-3e6f-43ec-9042-606ad7023533)

V7 doesn't have same structure or method so not an option as far as I can tell

**Reply by joshhanley** [reply]
@JonathanAspeling the `orchestra/testbench` package is a `require-dev` package for Livewire, so it shouldn't be installed when you install Livewire in your app. Can you confirm why your CI is trying to install it? Also Livewire V3 only supports Laravel 10, so not sure how you're managing to install it at all 🤔

**Reply by JonathanAspeling** [reply]
Thanks Josh! 

As far as I can tell it's not trying to install it, and because it is not installing it there's a piece in the Livewire\Features\SupportTesting\DuskTestable class. 

Lines 47 - 49
```
        if (isset($_SERVER['CI'])) {
            \Orchestra\Testbench\Dusk\Options::withoutUI();
        }
```
And then that fails because that isset return true when running the github CI actions. And then it can't find that dev depencancy and fails.

**Reply by joshhanley** [reply]
@JonathanAspeling ah right! What version of Livewire V3 are you using, because this has been fixed (a few versions ago)

https://github.com/livewire/livewire/blob/3535c5c9f4c4076d66656cedbdc084711054cd2b/src/Features/SupportTesting/DuskTestable.php#L48-L50

**Reply by JonathanAspeling** [reply]
Think we got Livewire V3 by requiring filament  🤔but apologies I'm still getting into the respective techs and even more so version/dependency management.

**Reply by JonathanAspeling** [reply]
Think I'm looking at the right place!

In my composer.lock file it's got the following so I think we're running: v3.0.0-beta.6

![image](https://github.com/livewire/livewire/assets/42679894/be14ddc5-74ce-431f-a3cd-4f23d27b0ad3)

But we are still on Laravel 9 so my recommendation to the team is that we get the V10 update sorted asap as I think we're starting to create a bit of a problem for ourselves haha

**Reply by joshhanley** [reply]
@JonathanAspeling no worries! I just checked, that was fixed in the Livewire beta, so it's worth doing an upgrade to ensure the latest Livewire version is installed (and that you're not still running on the beta).

You can run `composer update livewire/livewire` or if it is installed through Filament - assuming you're using the admin panel and not one of their packages - run `composer update filament/filament -W`. The `-W` will do an `update-with-all-dependencies` to ensure any of Filaments dependencies also get updated.

**Reply by joshhanley** [reply, marked as answer]
@JonathanAspeling ah yeah just saw your reply, yeah we had to drop Laravel 9 support due to Volt and a few other things. So yeah you will probably need to upgrade Laravel before you can upgrade from the Livewire beta.

**Reply by JonathanAspeling** [reply]
Great thank you for your time here! Added engineering item our side to usher in Laravel V10 for the app in question! Could be worse ways to find out it's time to upgrade 😜

**Reply by joshhanley** [reply]
@JonathanAspeling no worries! 🙂 Heh nice, yep 😆

**Comment by JonathanAspeling** [comment]
Resolved thanks so much! Discussion above if anyone else runs into our circumstance by chance! Issue our side :D
